src/utils/todo/todo.js

```diff
@@ -0,0 +1,120 @@
+import { TodoFilter } from '../../utils/todo/todoFilter.js';
+
+import { getTodos, setTodos } from '../../service/todo.service.js';
+
+import { DragAndDrop } from '../dnd.js';
+import { reorderList } from '../../store/dnd.store.js';
+import { renderList, renderTodo, updateSummary } from '../../ui/todo/todo.render.js';
+import { changeFilterList } from '../../ui/todo/todo.dom.js';
+
+import {
+  changeCompleteTodo,
+  updateTodoState,
+  addTodo,
+  setTodoState,
+  increaseIncomplete,
+} from '../../store/todo.store.js';
+
+export class TodoListApp extends TodoFilter {
+  todoState = {
+    complete: 0,
+    incomplete: 0,
+  };
+
+  useDnd = false;
+
+  constructor({ id, wrapper, options }) {
+    super();
+
+    this.$wrapper = wrapper;
+    this.id = 'todo_' + id;
+    this.todos = getTodos(this.id);
+    setTodoState.bind(this)();
+    if (options?.useDnd) this.useDnd = true;
+  }
+
+  submitTodo() {
+    this.$form.addEventListener('submit', (e) => {
+      e.preventDefault();
+
+      const $target = e.target;
+      const $input = $target.querySelector('.todo-input');
+      const value = $input.value;
+
+      if (!value) {
+        alert('값을 입력하세요');
+        return;
+      }
```

Comment:
p5. 지금은 공백도 정상적으로 입력되고 있는데, 공백만 입력될 때는 유효성 검사할 때 같이 걸러주면 어떨까요?

```suggestion
      const value = $input.value.trim();

      if (!value) {
        alert('값을 입력하세요');
        return;
      }
```